rio_tiler/utils.py

```diff
@@ -41,6 +41,12 @@ def aws_get_object(
     if not client:
         session = boto3_session()
         endpoint_url = os.environ.get("AWS_S3_ENDPOINT", None)
+        if endpoint_url is not None:
+            use_https = os.environ.get("AWS_HTTPS", "YES")
+            if use_https == "YES":
```

Comment:
```suggestion
            if use_https.upper() in ["YES", "TRUE", "ON"]:
```

This better align with GDAL bool